gwswlib/hydx.py

```diff
@@ -22,14 +22,16 @@ def import_csvline(cls, csvline):
             value = csvline[field["csvheader"]]
             datatype = field["type"]
 
-            if value == "":
-                value = None
-
             # set fields to defined data type and load into object
-            if value is not None:
-                setattr(instance, fieldname, datatype(value))
-            else:
+            if value is None or value == "":
                 setattr(instance, fieldname, None)
+            elif datatype == float and not isfloat(value):
```

Comment:
Nu check je "alleen" floats. Maar een `int("abc")` gaat nog steeds fout, bijvoorbeeld.

Op deze manier kan je alles afvangen (dit is dus een combi van de elif en de else in één keer:

    try:
        setattr(instance, fieldname, datatype(value))
    except ValueError:
        setattr(instance, fieldname, None)
        logger.error("%s in %s is not a %s: %r", fieldname, instance, datatype.__name__, value)

Ik zet als fallback, als het fout gaat, het attribute in ieder geval op None. Dan bestaat het in ieder geval wel. Niets is zo vervelend als een AttributeError als je erop rekent dat de attribute gewoon bestaat, vandaar.

Reply:
Ik heb het geprobeerd. Try except doet niet wat ik wil en maakt mijn None type een string van 'None', óf hij geeft een foutmelding terwijl die gewoon leeg is. Ik heb nog overwogen om de try except in de else toe te voegen, maar dat vind ik ook niet nodig, aangezien het in principe niet voorkomt dat...
In principe heb ik geen integers en anders kunnen deze gewoon worden toegevoegd. Zoveel datatypes zijn er nu ook weer niet toch?